src/service/packet_router.proto

```diff
@@ -52,10 +52,38 @@ message packet_router_register_v1 {
   bytes signature = 3;
 }
 
+// Session offer from the packet router to the gateway. If the gateway wants to
+// initiate a session key it should send a packet_router_session_init_v1 message
+// to the packet router using data in this offer to propose a session key.
+//
+// This message is sent by packet router after the register command is sent by
+// the gateway.
+message packet_router_session_offer_v1 { bytes data = 1; }
+
+// Initializes a session key with the packet router. The data in this init
+// message has to be from the latest session offer the gateway has received, the
+// session key is the public key that is to be used to verify packets by packet
+// router, and this message is to be signed by the private key of the gateway.
+// Once accepted the session key replaces the current (default is gateway) key
+// as the verifier used by the packet router for upilnk packets.
+//
+// The session key is valid for the length of the current stream or until a next
+// offer message is received from the packet router.
+//
+// Once the new verifier is set by the packet router, any signed packets that do
+// not pass verification with the active verifier key are dropped.
+message packet_router_session_init_v1 {
+  bytes gateway = 1;
+  bytes data = 2;
```

Comment:
"nonce" here as well.

Reply:
Fixed in [1e49483](https://github.com/helium/proto/pull/359/commits/1e4948358077027813cd81b70b30d20e67aa5901)